lib/neo4j/active_node/query/query_proxy.rb

```diff
@@ -186,11 +186,8 @@ def <<(other_node)
         #
         # @return [QueryProxy] A new QueryProxy
         def branch(&block)
-          if block
-            instance_eval(&block).query.proxy_as(self.model, identity)
-          else
-            fail LocalJumpError, 'no block given'
-          end
+          fail LocalJumpError, 'no block given' unless block
```

Comment:
This is a pet peeve of mine, so be warned ;)

From a standpoint of language, saying "fail unless X" is weird because most of the time you won't be failing.  It feels somewhat better to say "fail if not X" because it's supposed to be a rare case